src/core/reducers/errors.js

```diff
@@ -41,12 +44,23 @@ function getMessagesFromError(error) {
     });
 
     if (apiMessages.length) {
-      messages = apiMessages;
+      data = { ...data, messages: apiMessages };
     } else {
       log.warn('API error response did not contain any messages', error);
     }
+
+    if (
+      error.response.data.detail &&
+      // TODO: switch to checking for a code after
+      // https://github.com/mozilla/addons-server/issues/5122
+      error.response.data.detail === 'Signature has expired.'
+    ) {
+      // In this case, the only way to recover from the error is to
+      // refresh the page. It is impossible to fully avoid this state.
```

Comment:
Reloading the page is pretty disruptive. It's hard to know for sure that the user won't lose any work so I didn't want to do it automatically. I think hitting this use case will be uncommon so I don't think we need to optimize it too much.